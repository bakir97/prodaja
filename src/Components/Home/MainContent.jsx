import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import { connect } from "react-redux";
import Slika from "../../assets/global_shopping_experience.gif";
import Card from "../../Layout/Card/Card";
import withWidth from "@material-ui/core/withWidth";
import Styles from "./styles";
import SearchBarComponent from "./SearchBar/SearchBar";
import Switch from "./Switch";
import LayoutButtons from "./LayoutButtons";
import {
  getProizvodi,
  errorPostProizvod
} from "../../redux/actions/proizvodiActions";
import SnackBar from "../../Layout/SnackBar/SnackBar";
import Filters from "./Filters";
const styles = Styles;
class MainContent extends Component {
  state = {
    filters: false,
    layout: 12,
    filter: {
      minCijena: "",
      maxCijena: "",
      minTezina: "",
      maxTezina: ""
    }
  };
  componentDidUpdate() {
    if (this.props.width === "xs" && this.state.layout !== 12) {
      this.setState({ layout: 12 });
    }
  }
  componentWillUnmount() {
    this.props.errorPostProizvod(false);
  }
  componentDidMount() {
    this.props.getProizvodi();
  }
  cijenaFilters = filter => {
    if (typeof filter.minCijena === "string") {
      return this.setState({ minCijena: filter.minCijena });
    }
    if (typeof filter.maxCijena === "string") {
      return this.setState({ maxCijena: filter.maxCijena });
    }
    if (typeof filter.minTezina === "string") {
      return this.setState({ minTezina: filter.minTezina });
    }
    if (typeof filter.maxTezina === "string") {
      return this.setState({ maxTezina: filter.maxTezina });
    }
  };
  changeFiltersState = () => {
    this.setState(prevState => ({
      filters: !prevState.filters
    }));
  };
  changeLayout = number => {
    if (number !== this.state.layout) this.setState({ layout: number });
  };
  search = e => {
    const { minCijena, minTezina, maxCijena, maxTezina } = this.state;
    const podaci = {};
    podaci.search = e;
    podaci.minCijena = minCijena;
    podaci.maxCijena = maxCijena;
    podaci.minTezina = minTezina;
    podaci.maxTezina = maxTezina;
    this.props.getProizvodi(podaci);
  };
  snackRemove = e => {
    this.props.errorPostProizvod(e);
  };
  render() {
    const { classes, width, proizvodi, errorState, history } = this.props;
    const {
      filters,
      layout,
      minCijena,
      maxCijena,
      maxTezina,
      minTezina
    } = this.state;

    return (
      <Grid
        container
        alignContent="flex-start"
        alignItems="flex-start"
        justify="center"
        className={width === "xs" ? classes.rootMobile : classes.rootDesktop}
      >
        <Grid container justify="center">
          <Grid item container justify="center" xs={12}>
            <SearchBarComponent search={this.search} />
          </Grid>

          <Switch
            filters={filters}
            changeFiltersState={this.changeFiltersState}
          />
          {filters && (
            <Filters
              minCijena={minCijena}
              maxCijena={maxCijena}
              minTezina={minTezina}
              maxTezina={maxTezina}
              cijenaFilters={this.cijenaFilters}
            />
          )}
        </Grid>
        {width === "xs" ? null : (
          <LayoutButtons changeLayout={this.changeLayout} />
        )}
        {proizvodi.length > 0 ? (
          proizvodi.map(({ naslov, slike, cijena, tezina, _id }) => (
            <Grid key={_id} item container justify="center" xs={layout}>
              <Card
                history={history}
                id={_id}
                naslov={naslov}
                slika={slike[0]}
                cijena={cijena}
                tezina={tezina}
              />
            </Grid>
          ))
        ) : (
          <img className={classes.slika} src={Slika} alt="" />
        )}
        {errorState && (
          <SnackBar
            variantSnackBar="error"
            poruka={errorState}
            snackRemove={this.snackRemove}
          />
        )}
      </Grid>
    );
  }
}

const mapStateToProps = state => ({
  proizvodi: state.proizvodi.proizvodi,
  errorState: state.proizvodi.error
});

const mapDispatchToProps = {
  getProizvodi,
  errorPostProizvod
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withWidth()(withStyles(styles)(MainContent)));
