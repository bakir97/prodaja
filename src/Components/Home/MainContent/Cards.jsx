import React from "react";
import Card from "../../../Layout/Card/Card";
import { Grid } from "@material-ui/core";
const Cards = ({ proizvodi, layout, history }) => {
  return proizvodi.map(({ naslov, slike, cijena, tezina, _id }) => (
    <Grid key={_id} item container justify="center" xs={layout}>
      <Card
        history={history}
        id={_id}
        naslov={naslov}
        slika={slike[0]}
        cijena={cijena}
        tezina={tezina}
      />
    </Grid>
  ));
};

export default Cards;
