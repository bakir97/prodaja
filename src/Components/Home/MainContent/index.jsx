import React, { Component } from "react";
import { connect } from "react-redux";
import withWidth from "@material-ui/core/withWidth";
import {
  getProizvodi,
  errorPostProizvod
} from "../../../redux/actions/proizvodiActions";
import MainContentLayout from "./MainContent";
class MainContent extends Component {
  componentWillUnmount() {
    this.props.errorPostProizvod(false);
  }
  componentDidMount() {
    this.props.getProizvodi();
  }
  render() {
    return <MainContentLayout {...this.props} />;
  }
}

const mapStateToProps = state => ({
  proizvodi: state.proizvodi.proizvodi,
  errorState: state.proizvodi.error
});

const mapDispatchToProps = {
  getProizvodi,
  errorPostProizvod
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withWidth()(MainContent));
