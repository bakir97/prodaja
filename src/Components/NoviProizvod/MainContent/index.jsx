import React, { Component } from "react";
import MainContentLayout from "./MainContentLayout";

export class MainContent extends Component {
  state = {
    slike: [],
    oldPictures: []
  };
  componentDidUpdate() {
    if (
      Object.keys(this.props.newProizvodState).length > 0 &&
      this.state.slike.length > 0
    ) {
      this.setState({ slike: [] });
    }
    if (
      !this.props.match.params.id &&
      Object.keys(this.props.proizvod).length > 0
    ) {
      this.props.proizvodToRedux({});
      this.setState({ slike: [], oldPictures: [] });
    }
  }
  componentWillReceiveProps(nextProps) {
    if (
      this.props.proizvod !== nextProps.proizvod &&
      Object.keys(nextProps.proizvod).length > 0
    ) {
      this.setState({ oldPictures: nextProps.proizvod.slike });
    }
  }
  drop = files => {
    this.setState(prevState => ({
      slike: [...prevState.slike, ...files]
    }));
  };
  obrisi = id => {
    this.setState(prevState => ({
      slike: prevState.slike.filter(slika => slika.preview !== id)
    }));
  };
  obrisiOldPictures = id => {
    this.setState(prevState => ({
      oldPictures: prevState.oldPictures.filter(
        oldPictures => oldPictures !== id
      )
    }));
  };
  render() {
    return (
      <MainContentLayout
        {...this.props}
        {...this.state}
        drop={this.drop}
        obrisi={this.obrisi}
        obrisiOldPictures={this.obrisiOldPictures}
      />
    );
  }
}
export default MainContent;
