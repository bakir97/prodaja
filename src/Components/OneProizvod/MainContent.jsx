import React from "react";
import Style from "./mainStyle";
import PictureSlider from "./PictureSlider";
import { Grid, Typography, Button } from "@material-ui/core";
import AddShoppingCart from "@material-ui/icons/AddShoppingCart";

const Text = ({ children }) => {
  return (
    <Grid item xs={6} style={{ textAlign: "center" }}>
      <Typography variant="h5" style={{ fontFamily: "Kalam" }}>
        {children}
      </Typography>
    </Grid>
  );
};
const MainContent = ({ naslov, slike, opis, cijena, tezina }) => {
  return (
    <>
      <Grid item xs={12} style={Style.root}>
        <Typography variant="h4" style={Style.text}>
          {naslov}
        </Typography>
      </Grid>
      <Grid container justify="center" item xs={12}>
        <PictureSlider pictures={slike} />
      </Grid>
      <Typography variant="h5">{opis}</Typography>
      <Grid container justify="center" style={Style.container}>
        <Text>Tezina: {tezina} KG</Text>
        <Text>Cijena: {cijena} KM</Text>
        <Grid item xs={12} style={Style.buttonContainer}>
          <Button>
            <Typography
              variant="h5"
              style={{ ...Style.buttonText, ...Style.text }}
            >
              Dodaj u korpu
            </Typography>
            <AddShoppingCart
              style={{
                color: "#4286f4",
                marginBottom: 5
              }}
            />
          </Button>
        </Grid>
      </Grid>
    </>
  );
};

export default MainContent;
