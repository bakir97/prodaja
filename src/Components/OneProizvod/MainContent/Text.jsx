import React from "react";
import { Grid, Typography } from "@material-ui/core";
const style = {
  root: { textAlign: "center" },
  text: { fontFamily: "Kalam" }
};
const Text = ({ children }) => {
  return (
    <Grid item xs={6} style={style.root}>
      <Typography variant="h5" style={style.text}>
        {children}
      </Typography>
    </Grid>
  );
};

export default Text;
