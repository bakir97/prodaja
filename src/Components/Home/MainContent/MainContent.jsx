import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Slika from "../../../assets/global_shopping_experience.gif";
import Cards from "./Cards";
import Styles from "./styles";
import Header from "./Header";
import SnackBar from "../../../Layout/SnackBar/SnackBar";
const styles = Styles;
class MainContent extends Component {
  state = {
    layout: 12
  };
  componentDidUpdate() {
    if (this.props.width === "xs" && this.state.layout !== 12) {
      this.setState({ layout: 12 });
    }
  }
  changeLayout = number => {
    if (number !== this.state.layout) this.setState({ layout: number });
  };

  snackRemove = e => {
    this.props.errorPostProizvod(e);
  };
  render() {
    const {
      classes,
      width,
      proizvodi,
      errorState,
      history,
      getProizvodi
    } = this.props;
    const { layout } = this.state;

    return (
      <Grid
        container
        alignContent="flex-start"
        alignItems="flex-start"
        justify="center"
        className={width === "xs" ? classes.rootMobile : classes.rootDesktop}
      >
        <Header
          width={width}
          getProizvodi={getProizvodi}
          changeLayout={this.changeLayout}
        />
        {proizvodi.length > 0 ? (
          <Cards proizvodi={proizvodi} layout={layout} history={history} />
        ) : (
          <img className={classes.slika} src={Slika} alt="" />
        )}
        {errorState && (
          <SnackBar
            variantSnackBar="error"
            poruka={errorState}
            snackRemove={this.snackRemove}
          />
        )}
      </Grid>
    );
  }
}

export default withStyles(styles)(MainContent);
