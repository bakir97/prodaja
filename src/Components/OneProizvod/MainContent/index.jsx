import React from "react";
import Style from "./mainStyle";
import PictureSlider from "../PictureSlider";
import { Grid, Typography, Button } from "@material-ui/core";
import AddShoppingCart from "@material-ui/icons/AddShoppingCart";
import Text from "./Text";
const MainContent = ({ naslov, slike, opis, cijena, tezina }) => {
  return (
    <>
      <Grid item xs={12} style={Style.root}>
        <Typography variant="h4" style={Style.text}>
          {naslov}
        </Typography>
      </Grid>
      <Grid container justify="center" item xs={12}>
        <PictureSlider pictures={slike} />
      </Grid>
      <Typography variant="h5">{opis}</Typography>
      <Grid container justify="center" style={Style.container}>
        <Text>Tezina: {tezina} KG</Text>
        <Text>Cijena: {cijena} KM</Text>
        <Grid item xs={12} style={Style.buttonContainer}>
          <Button>
            <Typography
              variant="h5"
              style={{ ...Style.buttonText, ...Style.text }}
            >
              Dodaj u korpu
            </Typography>
            <AddShoppingCart style={Style.shoppingCart} />
          </Button>
        </Grid>
      </Grid>
    </>
  );
};

export default MainContent;
