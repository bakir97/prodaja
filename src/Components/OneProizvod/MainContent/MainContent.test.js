import MainContent from "./index";
import Text from "./Text";
describe("MainContent", () => {
  const wrapper = shallow(<MainContent />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
describe("Text", () => {
  const wrapper = shallow(<Text />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
