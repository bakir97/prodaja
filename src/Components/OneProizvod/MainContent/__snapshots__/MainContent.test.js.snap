// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent should render 1`] = `
<Fragment>
  <WithStyles(Grid)
    item={true}
    style={
      Object {
        "marginBottom": "2rem",
        "textAlign": "center",
      }
    }
    xs={12}
  >
    <WithStyles(Typography)
      style={
        Object {
          "WebkitBackgroundClip": "text",
          "background": "linear-gradient(to right, #373b44, #4286f4)",
          "color": "transparent",
        }
      }
      variant="h4"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    xs={12}
  >
    <WithStyles(TextMobileStepper) />
  </WithStyles(Grid)>
  <WithStyles(Typography)
    variant="h5"
  />
  <WithStyles(Grid)
    container={true}
    justify="center"
    style={
      Object {
        "marginTop": "2rem",
        "padding": "0 1rem",
      }
    }
  >
    <Text>
      Tezina: 
       KG
    </Text>
    <Text>
      Cijena: 
       KM
    </Text>
    <WithStyles(Grid)
      item={true}
      style={
        Object {
          "marginTop": "2rem",
          "textAlign": "center",
        }
      }
      xs={12}
    >
      <WithStyles(Button)>
        <WithStyles(Typography)
          style={
            Object {
              "WebkitBackgroundClip": "text",
              "background": "linear-gradient(to right, #373b44, #4286f4)",
              "color": "transparent",
              "fontFamily": "Kalam",
              "marginRight": "1rem",
              "paddingBottom": 0,
            }
          }
          variant="h5"
        >
          Dodaj u korpu
        </WithStyles(Typography)>
        <pure(AddShoppingCartIcon)
          style={
            Object {
              "color": "#4286f4",
              "marginBottom": 5,
            }
          }
        />
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Fragment>
`;

exports[`Text should render 1`] = `
<WithStyles(Grid)
  item={true}
  style={
    Object {
      "textAlign": "center",
    }
  }
  xs={6}
>
  <WithStyles(Typography)
    style={
      Object {
        "fontFamily": "Kalam",
      }
    }
    variant="h5"
  />
</WithStyles(Grid)>
`;
